shumai/tensor/tensor_ops.ts

```diff
@@ -0,0 +1,6 @@
+import type { Tensor } from './tensor'
+
+export function softmax(tensor: Tensor, axis: number): Tensor {
+  const exp = tensor.exp()
+  return exp.div(exp.sum([axis], true))
+}
```

Comment:
I really appreciate the organizational design choice here to add this in a separate file, as it'll keep a clean delineation of where the logic for non-generated tensor ops lives.